### PR TITLE
feat: refactor `Pfi`

### PR DESCRIPTION
**Pull request by ethan-tbd**
closes https://github.com/TBD54566975/didpay/issues/153

### CODE REVIEW

File: ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "1510"
+   LastUpgradeVersion = "1430"
```
Comment (wesbillman) [inline ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:3]: What is happening here @ethan-tbd !!!! :)

File: ios/Podfile.lock [hunk 2]
```diff
@@ -134,9 +134,9 @@ SPEC CHECKSUMS:
...
   shared_preferences_foundation: b4c3b4cddf1c21f02770737f147a3f5da9d39695
-  url_launcher_ios: 6116280ddcfe98ab8820085d8d76ae7449447586
-  webview_flutter_wkwebview: be0f0d33777f1bfd0c9fdcb594786704dbf65f36
+  url_launcher_ios: bbd758c6e7f9fd7b5b1d4cde34d2b95fcce5e812
+  webview_flutter_wkwebview: 4f3e50f7273d31e5500066ed267e3ae4309c5ae4
 
 PODFILE CHECKSUM: 819463e6a0290f5a72f145ba7cde16e8b6ef0796
 
-COCOAPODS: 1.15.2
+COCOAPODS: 1.13.0
```
Comment (mistermoe) [inline ios/Podfile.lock:142]: hmmm down-versioning?

File: lib/features/did_qr/did_qr.dart
Comment (mistermoe) [inline lib/features/did_qr/did_qr.dart:1]: this is just for pfi dids yeah?

File: lib/features/home/home_page.dart [hunk 3]
```diff
@@ -20,6 +23,8 @@ class HomePage extends HookConsumerWidget {
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final txns = ref.watch(transactionsProvider);
+    final pfis = ref.watch(pfisProvider);
+
     // TODO(ethan-tbd): get balance from pfi, https://github.com/TBD54566975/didpay/issues/109
     final accountBalance = CurrencyUtil.formatFromDouble(0);
```
Comment (mistermoe) [inline lib/features/home/home_page.dart:29]: not important for this PR but we should switch to using [this decimal lib](https://pub.dev/packages/decimal)

File: lib/features/home/home_page.dart [hunk 4]
```diff
@@ -104,38 +120,39 @@ class HomePage extends HookConsumerWidget {
...
+                        onPressed: () {
+                          Navigator.of(context).push(
+                            MaterialPageRoute(
+                              builder: (context) =>
+                                  const DepositPage(rfqState: RfqState()),
+                            ),
+                          );
+                        },
+                        child: Text(Loc.of(context).deposit),
+                      ),
```
Comment (mistermoe) [inline lib/features/home/home_page.dart:138]: hmm would this make more sense to pull out of `_buildAccountBalance`?

File: lib/features/pfis/add_pfi_confirmation_page.dart [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    return Scaffold(
+      body: SafeArea(
+        child: addPfiState.value.when(
+          data: (pfi) => SuccessPage(text: Loc.of(context).pfiAdded),
+          loading: () => PendingPage(text: Loc.of(context).addingPfi),
+          error: (error, _) => Text('Error: $error'),
+        ),
+      ),
+    );
+  }
```
Comment (mistermoe) [inline lib/features/pfis/add_pfi_confirmation_page.dart:39]: image

also not important in this PR but wonder if this is necessary? or if we can just redirect back to where the user came from?

File: lib/features/pfis/pfi.dart
Comment (mistermoe) [inline lib/features/pfis/pfi.dart:1]: i thiiink we decided to stick with just `did`? but maybe we changed our minds i can't remember why

File: lib/features/pfis/pfis_service.dart [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:web5/web5.dart';
+
+final pfisServiceProvider = Provider((_) => PfisService());
+
+class PfisService {
+  Future<Pfi> createPfi(String input) async {
+    final did = _parseDid(input);
+    final res = await _resolveDid(did);
+    final didDocument = _getDidDocument(res);
```
Comment (mistermoe) [inline lib/features/pfis/pfis_service.dart:11]: any reason to split all these out into separate functions? do you foresee us re-using these anywhere aside from right here?

File: lib/features/send/send_did_page.dart
Comment (mistermoe) [inline lib/features/send/send_did_page.dart:1]: what's sending a did?

**Comment by mistermoe** [comment]
an issue we can create after this is merged:

probably can prevent this from being a textarea and make it just an input (e.g. can't press enter)


https://github.com/TBD54566975/didpay/assets/4887440/d2a822ed-cf89-4c9b-b68a-b437b2cf83f3

**Comment by mistermoe** [comment]
is the expected behavior for this to spin foreva?



https://github.com/TBD54566975/didpay/assets/4887440/fb129857-b491-4837-a247-0a98e167180c

**Comment by mistermoe** [comment]
is deposit screen supposed to load?

https://github.com/TBD54566975/didpay/assets/4887440/1ffe70e1-bb9d-43a2-bbef-fbd45ad5a0c0

File: ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme (hunk 1 above)
Comment (ethan-tbd) [reply ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:3]: hey these changes came from @mistermoe 😂

File: lib/features/did_qr/did_qr.dart
Comment (ethan-tbd) [reply lib/features/did_qr/did_qr.dart:1]: this is also being used for `SendDidPage`, which is where we would eventually use DAPs / these did qr codes

File: ios/Podfile.lock (hunk 2 above)
Comment (ethan-tbd) [reply ios/Podfile.lock:142]: yea idk what this is from, prob best to leave out of this pr

File: lib/features/home/home_page.dart (hunk 4 above)
Comment (ethan-tbd) [reply lib/features/home/home_page.dart:138]: yep, can make that happen

File: lib/features/pfis/add_pfi_confirmation_page.dart (hunk 5 above)
Comment (ethan-tbd) [reply lib/features/pfis/add_pfi_confirmation_page.dart:39]: so after the pfi has been successfully added, it goes back to `AppTabs` right away?

File: lib/features/pfis/pfi.dart
Comment (ethan-tbd) [reply lib/features/pfis/pfi.dart:1]: gotcha, didn't know if we had talked about it. just a `did` works perfect!

File: lib/features/pfis/pfis_service.dart (hunk 6 above)
Comment (ethan-tbd) [reply lib/features/pfis/pfis_service.dart:11]: nahh i was just braindead haha

File: lib/features/send/send_did_page.dart
Comment (ethan-tbd) [reply lib/features/send/send_did_page.dart:1]: name of this page should change.. but this is just a continuation of the send page (where you enter the recipient's did)

**Comment by ethan-tbd** [comment]
> is the expected behavior for this to spin foreva?

no, a future pr will add polling and a timeout to this

**Comment by ethan-tbd** [comment]
> is deposit screen supposed to load?

will add a fix for this, i think we made some changes to `TbdexHttpClient` when doing some local testing

**Pull request merged**